flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -741,6 +741,9 @@ public boolean isSysoutLoggingEnabled() {
 	}
 
 	public GlobalJobParameters getGlobalJobParameters() {
+		if (globalJobParameters == null) {
+			globalJobParameters = new GlobalJobParameters();
```

Comment:
Wouldn't it be better to initialize `globalJobParameters` in the constructor of the `ExecutionConfig` and then introduce a not null check in the `setGlobalJobParameters` method? That way, our invariant would be that this field is never `null`.